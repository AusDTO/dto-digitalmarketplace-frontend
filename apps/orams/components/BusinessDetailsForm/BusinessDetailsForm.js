/* eslint-disable */

import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { Form } from 'react-redux-form'
import AUpageAlert from '@gov.au/page-alerts/lib/js/react.js'

import { loadProfile } from 'orams/actions/profileActions'
import { hideNav } from 'orams/actions/editPriceActions'

import { required, limitNumbers, validLinks, validABN } from 'shared/validators'
import Layout from 'shared/Layout'
import BaseForm from 'shared/form/BaseForm'
import ErrorBox from 'shared/form/ErrorBox'
import Textarea from 'shared/form/Textarea'
import Textfield from 'shared/form/Textfield'
import formProps from 'shared/form/formPropsSelector'
import LoadingButton from 'shared/LoadingButton/LoadingButton'

import './BusinessDetailsForm.scss'

class BusinessDetailsForm extends BaseForm {
  static propTypes = {
    action: PropTypes.string,
    csrf_token: PropTypes.string
  }

  componentDidMount() {
    this.props.hideNav(null)
    this.props.loadProfileData(this.props.supplierCode, this.props.form.model)
  }

  onAdd(e) {
    e.preventDefault()
    const { model, addAddress, businessDetailsForm } = this.props
    addAddress(model, businessDetailsForm.addresses || {})
  }

  onRemove(id, e) {
    e.preventDefault()
    const { model, removeAddress } = this.props
    removeAddress(model, id)
  }

  render() {
    const {
      submitClicked,
      action,
      csrf_token,
      model,
      children,
      handleSubmit,
      businessDetailsForm,
      profileUpdated,
      currentlySending
    } = this.props
    let hasFocused = false
    const setFocus = e => {
      if (!hasFocused) {
        hasFocused = true
        e.focus()
      }
    }

    return (
      <Layout>
        <header>
          <h1 className="au-display-xl" tabIndex="-1">
            Check your business details
          </h1>
        </header>
        <article role="main">
          {profileUpdated
            ? <AUpageAlert as="success">
                <h4>Profile updated</h4>
              </AUpageAlert>
            : ''}
          <ErrorBox model={model} setFocus={setFocus} submitClicked={submitClicked} />
          <Form
            model={model}
            action={action}
            id="BusinessDetails__create"
            validateOn="submit"
            onSubmit={data => handleSubmit(data)}
          >
            {csrf_token && <input type="hidden" name="csrf_token" id="csrf_token" value={csrf_token} />}

            <Textfield
              model={`${model}.name`}
              name="name"
              id="name"
              htmlFor="name"
              label="Business name"
              validators={{ required }}
              messages={{
                required: 'Business name is required'
              }}
            />

            <Textfield
              model={`${model}.abn`}
              name="abn"
              id="abn"
              htmlFor="abn"
              label="ABN"
              description={<span>You need an Australian Business Number to do business in Australia.</span>}
              messages={{
                validABN: 'ABN is required and must match a valid ABN as listed on the Australian Business Register'
              }}
              validators={{ validABN }}
            />

            <Textarea
              model={`${model}.summary`}
              name="summary"
              id="summary"
              controlProps
              label="Summary"
              description="3-4 sentences that describe your business. This can be seen by all government buyers."
            />

            <Textfield
              model={`${model}.website`}
              name="website"
              id="website"
              htmlFor="website"
              label="Website URL"
              description="Provide a link to your website beginning with http"
              messages={{
                required: 'You must provide a website link beginning with http',
                validLinks: 'Links provided must begin with http'
              }}
              validators={{ required, validLinks }}
            />

            <Textfield
              model={`${model}.linkedin`}
              name="linkedin"
              id="linkedin"
              htmlFor="linkedin"
              label="LinkedIn URL (optional)"
              description="Provide a LinkedIn URL beginning with http"
              messages={{
                validLinks: 'Links provided must begin with http'
              }}
              validators={{ validLinks }}
            />

            <Textfield
              model={`${model}.address_address_line`}
              name="address.address_line"
              id="address_line"
              htmlFor="address_line"
              label="Primary Address"
              description="Principal place of business."
              messages={{
                required: 'You must provide an address'
              }}
              validators={{ required }}
            />

            <Textfield
              model={`${model}.address_suburb`}
              name="address.suburb"
              id="suburb"
              htmlFor="suburb"
              label="Suburb"
              messages={{
                required: 'You must provide a suburb'
              }}
              validators={{ required }}
            />
            <Textfield
              model={`${model}.address_state`}
              name="address.state"
              id="state"
              htmlFor="state"
              label="State"
              messages={{
                required: 'You must provide a state'
              }}
              validators={{ required }}
            />
            <Textfield
              model={`${model}.address_postal_code`}
              name="address.postal_code"
              id="postal_code"
              htmlFor="postal_code"
              label="Postcode"
              maxLength="4"
              messages={{
                required: 'You must provide a postal code. ',
                limitNumbers: 'Postal codes must be four digits long and only numbers.'
              }}
              validators={{ required, limitNumbers: limitNumbers(4) }}
            />
            <div>
              {businessDetailsForm.addresses &&
                Object.keys(businessDetailsForm.addresses)
                  .filter(value => {
                    return value > 0 && businessDetailsForm.addresses[value] !== null
                  })
                  .map((key, i) => {
                    return (
                      <div className="address-wrapper" key={key}>
                        <hr className="hr" />
                        <div className="row">
                          <div className="col-xs-8 col-sm-10">
                            <h3 className="au-display-lg">Additional address</h3>
                          </div>
                          <div className="col-xs-4 col-sm-2">
                            <a href="#" onClick={this.onRemove.bind(this, key)}>
                              Remove
                            </a>
                          </div>
                        </div>
                        <div className="row">
                          <div className="col-xs-12 col-sm-10">
                            <Textfield
                              model={`${model}.addresses.${key}.address_line`}
                              name={`address_line-${key}`}
                              id={`address_line-${key}`}
                              htmlFor={`address_line-${key}`}
                              label="Address"
                              messages={{
                                required: 'You must provide address'
                              }}
                              validators={{ required }}
                            />
                            <Textfield
                              model={`${model}.addresses.${key}.suburb`}
                              name={`suburb-${key}`}
                              id={`suburb-${key}`}
                              htmlFor={`suburb-${key}`}
                              label="Suburb"
                              messages={{
                                required: 'You must provide a suburb'
                              }}
                              validators={{ required }}
                            />
                            <Textfield
                              model={`${model}.addresses.${key}.state`}
                              name={`state-${key}`}
                              id={`state-${key}`}
                              htmlFor={`state-${key}`}
                              label="State"
                              messages={{
                                required: 'You must provide a state'
                              }}
                              validators={{ required }}
                            />
                            <Textfield
                              model={`${model}.addresses.${key}.postal_code`}
                              name={`postal_code-${key}`}
                              id={`postal_code-${key}`}
                              htmlFor={`postal_code-${key}`}
                              label="Postcode"
                              maxLength="4"
                              messages={{
                                required: 'You must provide a postal code. ',
                                limitNumbers: 'Postal codes must be four digits long and only numbers.'
                              }}
                              validators={{ required, limitNumbers: limitNumbers(4) }}
                            />
                          </div>
                        </div>
                      </div>
                    )
                  })}
            </div>

            {children}

            {currentlySending
              ? <LoadingButton />
              : <button type="submit" className="au-btn">
                  Update profile
                </button>}
          </Form>
        </article>
      </Layout>
    )
  }
}

const mapStateToProps = state => {
  return {
    ...formProps(state, 'businessDetailsForm'),
    profileUpdated: state.app.profileUpdated,
    currentlySending: state.app.currentlySending,
    supplierCode: state.app.supplierCode
  }
}

const mapDispatchToProps = dispatch => {
  return {
    hideNav: bool => dispatch(hideNav(bool)),
    loadProfileData: (supplierCode, form) => dispatch(loadProfile(supplierCode, form))
  }
}

export { Textfield, mapStateToProps, BusinessDetailsForm as Form }

export default connect(mapStateToProps, mapDispatchToProps)(BusinessDetailsForm)
