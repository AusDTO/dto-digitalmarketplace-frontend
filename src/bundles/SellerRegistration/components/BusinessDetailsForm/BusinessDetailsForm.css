.heading {
	margin:0;
}

.remove-button {
	float:right;
	margin: 0;
}

.address-wrapper {
	margin-top:2rem;
}

.hr {
	margin-bottom:2rem;
}

.footer {
	margin:0;
}
