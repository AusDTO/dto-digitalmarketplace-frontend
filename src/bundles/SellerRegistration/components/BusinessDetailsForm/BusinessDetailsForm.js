import React from 'react';
import {connect} from 'react-redux';
import { Form, Control } from 'react-redux-form';

import {required} from '../../../../validators';

import Layout from '../../../../shared/Layout';

import BaseForm     from '../../../../shared/form/BaseForm';
import SubmitForm   from '../../../../shared/form/SubmitForm';
import ErrorBox     from '../../../../shared/form/ErrorBox';
import Textarea     from '../../../../shared/form/Textarea';
import Textfield    from '../../../../shared/form/Textfield';
import formProps    from '../../../../shared/reduxModules/formPropsSelector';


class BusinessDetailsForm extends BaseForm {

    static propTypes = {
        action: React.PropTypes.string,
        csrf_token: React.PropTypes.string,
        form: React.PropTypes.object.isRequired,
        returnLink: React.PropTypes.string
    }

    render() {
        const {action, csrf_token, model, returnLink, form, title, buttonText, seller_registration, children, onSubmit } = this.props;
        return (
            <Layout>
                <header>
                    <h1 tabIndex="-1">Check your details</h1>
                </header>
                <article role="main">
                    <ErrorBox focusOnMount={true} model={model}/>
                    <Form model={model}
                          action={action}
                          method="post"
                          id="BusinessDetails__create"
                          valid={form.valid}
                          component={SubmitForm}
                          onCustomSubmit={onSubmit}
                    >
                        {csrf_token && (
                            <input type="hidden" name="csrf_token" id="csrf_token" value={csrf_token}/>
                        )}

                        {(seller_registration === true) &&
                            <div>
                                <Textfield
                                  model={`${model}.name`}
                                  name="name"
                                  id="name"
                                  htmlFor="name"
                                  label="Business Name"
                                  description="As you would like it shown on the Marketplace"
                                  validators={{ required }}
                                  messages={{
                                    required: 'Business name is required',
                                  }} 
                                />

                                <Textfield
                                  model={`${model}.abn`}
                                  name="abn"
                                  id="abn"
                                  htmlFor="abn"
                                  label="ABN"
                                  validators={{ required }}
                                  messages={{
                                    required: 'ABN is required',
                                  }}
                                />
                            </div>
                        }

                        <Textarea
                            model={`${model}.summary`}
                            name="summary"
                            id="summary"
                            controlProps={{limit: 50}}
                            label="Summary"
                            description="3-4 sentences that describe your business. This is visible to all viewers without a login."
                            messages={{
                                required: 'You must provide a seller summary'
                            }}
                            validators={{required}}
                        />

                        <Textfield
                            model={`${model}.website`}
                            name="website"
                            id="website"
                            htmlFor="website"
                            label="Website URL"
                            messages={{
                                required: 'You must provide a website URL'
                            }}
                            validators={{required}}
                        />

                        <Textfield
                            model={`${model}.linkedin`}
                            name="linkedin"
                            id="linkedin"
                            htmlFor="linkedin"
                            label="LinkedIn URL (optional)"
                        />

                        <Textfield
                            model={`${model}.address.address_line`}
                            name="address.address_line"
                            id="address_line"
                            htmlFor="address_line"
                            label="Address"
                            description="Principal place of business"
                            messages={{
                                required: 'You must provide an address'
                            }}
                            validators={{required}}
                        />

                        <Textfield
                            model={`${model}.address.suburb`}
                            name="address.suburb"
                            id="suburb"
                            htmlFor="suburb"
                            label="Suburb"
                            messages={{
                                required: 'You must provide a suburb'
                            }}
                            validators={{required}}
                        />
                        <Textfield
                            model={`${model}.address.state`}
                            name="address.state"
                            id="state"
                            htmlFor="state"
                            label="State"
                            messages={{
                                required: 'You must provide a state'
                            }}
                            validators={{required}}
                        />
                        <Textfield
                            model={`${model}.address.postal_code`}
                            name="address.postal_code"
                            id="postal_code"
                            htmlFor="postal_code"
                            label="Postcode"
                            maxLength="4"
                            messages={{
                                required: 'You must provide a postal code'
                            }}
                            validators={{required}}
                            pattern="[0-9]{4}"
                        />

                        <Control.checkbox
                          model={`${model}.interstate`}
                          id="interstate"
                          name="interstate"
                        />
                        <label htmlFor="interstate">
                            Able to work regionally or interstate
                        </label>

                        {children}

                        <input type="submit" value={buttonText} role="button"/>
                    </Form>
                    {returnLink && <a href={returnLink}>Return without saving</a>}
                </article>
            </Layout>
        )
    }
}

BusinessDetailsForm.defaultProps = {
  buttonText: 'Update profile',
  title: 'Tell us about your business'
}

const mapStateToProps = (state) => {
    return {
        returnLink: state.businessDetailsForm && state.businessDetailsForm.returnLink,
        ...formProps(state, 'businessDetailsForm')
    }
}

export {
    Textfield,
    mapStateToProps,
    BusinessDetailsForm as Form
}

export default connect(mapStateToProps)(BusinessDetailsForm);
