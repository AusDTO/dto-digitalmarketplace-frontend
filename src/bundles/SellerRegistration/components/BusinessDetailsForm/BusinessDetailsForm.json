{
  "businessDetailsForm": {
    "name": "SupCo Pty. Ltd.",
    "summary": "",
    "linkedin": "www.linked.in",
    "website": "www.com",
    "addresses": [{
      "address_line": "Level 5, 77 Devon Street",
      "suburb": "Surry Hills",
      "state": "NSW",
      "postal_code": "2010",
      "country": "Australia"
    }]
  },
  "form_options": {
    "errors": {
      "summary": {
        "required": true
      }
    }
  }
}
