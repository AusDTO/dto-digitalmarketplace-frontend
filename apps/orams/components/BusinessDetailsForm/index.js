import BusinessDetailsForm from './BusinessDetailsForm'

export default BusinessDetailsForm
