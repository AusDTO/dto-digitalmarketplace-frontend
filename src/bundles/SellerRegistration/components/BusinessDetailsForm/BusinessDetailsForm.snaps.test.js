// FIXME This is a workaround.
// @see https://github.com/facebook/react/issues/7386
jest.mock('react-dom');

import React from 'react'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'

import BusinessDetailsForm, { Textfield } from './BusinessDetailsForm'
import sampleState from './BusinessDetailsForm.json'
import createStore from '../../redux/create'

test('BusinessDetailsForm renders', () => {
  let store = createStore(Object.assign({}, { _serverContext: {} }))
  const component = renderer.create(
  	<Provider store={store}>
    	<BusinessDetailsForm />
    </Provider>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('BusinessDetailsForm renders with form_options', () => {
  const form_options = {
    csrf_token: 'sometoken',
    action: '/foo/bar'
  }
  let store = createStore(Object.assign({}, { _serverContext: {}, form_options }))
  const component = renderer.create(
    <Provider store={store}>
      <BusinessDetailsForm />
    </Provider>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test.skip('BusinessDetailsForm renders with populated fields', () => {
  let store = createStore(Object.assign({}, { _serverContext: {} }, sampleState))
  const component = renderer.create(
    <Provider store={store}>
      <BusinessDetailsForm />
    </Provider>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

