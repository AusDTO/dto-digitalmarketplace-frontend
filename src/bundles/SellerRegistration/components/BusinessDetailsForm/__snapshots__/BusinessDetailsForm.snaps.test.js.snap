// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BusinessDetailsForm renders 1`] = `
<div
  className="layout"
>
  <header>
    <h1
      tabIndex="-1"
    >
      Tell us about your business
    </h1>
  </header>
  <article
    role="main"
  >
    <form
      action={undefined}
      id="BusinessDetails__create"
      method="post"
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <div
        className="field"
      >
        <label
          className="question-heading"
          htmlFor="name"
        >
          Business name
        </label>
        <p
          className="hint"
          id="name-hint"
        >
          As you would like it shown on the Digital Marketplace.
        </p>
        <input
          aria-describedby="name-hint"
          className=""
          disabled={false}
          id="name"
          maxLength={undefined}
          name="name"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          pattern={undefined}
          readOnly={undefined}
          type="text"
          value=""
        />
      </div>
      <div
        className="field"
      >
        <label
          className="question-heading"
          htmlFor="abn"
        >
          ABN
        </label>
        <p
          className="hint"
          id="abn-hint"
        >
          <span>
            You need an Australian Business Number to do business in Australia. 
            <a
              href="https://abr.gov.au/For-Business,-Super-funds---Charities/Applying-for-an-ABN/Apply-for-an-ABN/"
              rel="external"
              target="_blank"
            >
              Apply for an ABN here.
            </a>
          </span>
        </p>
        <input
          aria-describedby="abn-hint"
          className=""
          disabled={false}
          id="abn"
          maxLength={undefined}
          name="abn"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          pattern={undefined}
          readOnly={false}
          type="text"
          value=""
        />
      </div>
      <div
        className="field"
      >
        <label
          className="question-heading"
          htmlFor="summary"
        >
          Summary
        </label>
        <p
          className="hint"
          id="summary-hint"
        >
          3-4 sentences that describe your business. This can be seen by all Digital Marketplace visitors without signing in.
        </p>
        <div>
          <textarea
            aria-describedby="summary-hint"
            className="under-50"
            defaultValue=""
            id="summary"
            name="summary"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
          />
          <span
            className="word-count-counter"
          >
            50 words remaining
          </span>
        </div>
      </div>
      <div
        className="field"
      >
        <label
          className="question-heading"
          htmlFor="website"
        >
          Website URL
        </label>
        <p
          className="hint"
          id="website-hint"
        >
          Provide a link to your website beginning with http
        </p>
        <input
          aria-describedby="website-hint"
          className=""
          disabled={false}
          id="website"
          maxLength={undefined}
          name="website"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          pattern={undefined}
          readOnly={undefined}
          type="text"
          value=""
        />
      </div>
      <div
        className="field"
      >
        <label
          className="question-heading"
          htmlFor="linkedin"
        >
          LinkedIn URL (optional)
        </label>
        <p
          className="hint"
          id="linkedin-hint"
        >
          Provide a LinkedIn URL beginning with http
        </p>
        <input
          aria-describedby="linkedin-hint"
          className=""
          disabled={false}
          id="linkedin"
          maxLength={undefined}
          name="linkedin"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          pattern={undefined}
          readOnly={undefined}
          type="text"
          value=""
        />
      </div>
      <div
        className="field"
      >
        <label
          className="question-heading"
          htmlFor="address_line"
        >
          Primary Address
        </label>
        <p
          className="hint"
          id="address_line-hint"
        >
          Principal place of business.
        </p>
        <input
          aria-describedby="address_line-hint"
          className=""
          disabled={false}
          id="address_line"
          maxLength={undefined}
          name="address.address_line"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          pattern={undefined}
          readOnly={undefined}
          type="text"
          value=""
        />
      </div>
      <div
        className="field"
      >
        <label
          className="question-heading"
          htmlFor="suburb"
        >
          Suburb
        </label>
        <input
          aria-describedby={undefined}
          className=""
          disabled={false}
          id="suburb"
          maxLength={undefined}
          name="address.suburb"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          pattern={undefined}
          readOnly={undefined}
          type="text"
          value=""
        />
      </div>
      <div
        className="field"
      >
        <label
          className="question-heading"
          htmlFor="state"
        >
          State
        </label>
        <input
          aria-describedby={undefined}
          className=""
          disabled={false}
          id="state"
          maxLength={undefined}
          name="address.state"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          pattern={undefined}
          readOnly={undefined}
          type="text"
          value=""
        />
      </div>
      <div
        className="field"
      >
        <label
          className="question-heading"
          htmlFor="postal_code"
        >
          Postcode
        </label>
        <input
          aria-describedby={undefined}
          className=""
          disabled={false}
          id="postal_code"
          maxLength="4"
          name="address.postal_code"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          pattern={undefined}
          readOnly={undefined}
          type="text"
          value=""
        />
      </div>
      <div>
        <p
          styleName="footer"
        >
          More offices?
        </p>
        <button
          className="button-secondary"
          onClick={[Function]}
          type="submit"
        >
          Add another address
        </button>
      </div>
      <div
        styleName="nav-row"
      >
        <button
          styleName="nav-button"
          type="submit"
        >
          Update profile
        </button>
        <button
          className="save-button"
          onClick={[Function]}
          styleName="nav-button"
        >
          Save and finish later
        </button>
        <button
          className="button-secondary"
          onClick={[Function]}
          styleName="nav-link skip-button"
        >
          Skip for now
        </button>
      </div>
    </form>
  </article>
</div>
`;

exports[`BusinessDetailsForm renders with form_options 1`] = `
<div
  className="layout"
>
  <header>
    <h1
      tabIndex="-1"
    >
      Tell us about your business
    </h1>
  </header>
  <article
    role="main"
  >
    <form
      action="/foo/bar"
      id="BusinessDetails__create"
      method="post"
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <input
        id="csrf_token"
        name="csrf_token"
        type="hidden"
        value="sometoken"
      />
      <div
        className="field"
      >
        <label
          className="question-heading"
          htmlFor="name"
        >
          Business name
        </label>
        <p
          className="hint"
          id="name-hint"
        >
          As you would like it shown on the Digital Marketplace.
        </p>
        <input
          aria-describedby="name-hint"
          className=""
          disabled={false}
          id="name"
          maxLength={undefined}
          name="name"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          pattern={undefined}
          readOnly={undefined}
          type="text"
          value=""
        />
      </div>
      <div
        className="field"
      >
        <label
          className="question-heading"
          htmlFor="abn"
        >
          ABN
        </label>
        <p
          className="hint"
          id="abn-hint"
        >
          <span>
            You need an Australian Business Number to do business in Australia. 
            <a
              href="https://abr.gov.au/For-Business,-Super-funds---Charities/Applying-for-an-ABN/Apply-for-an-ABN/"
              rel="external"
              target="_blank"
            >
              Apply for an ABN here.
            </a>
          </span>
        </p>
        <input
          aria-describedby="abn-hint"
          className=""
          disabled={false}
          id="abn"
          maxLength={undefined}
          name="abn"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          pattern={undefined}
          readOnly={false}
          type="text"
          value=""
        />
      </div>
      <div
        className="field"
      >
        <label
          className="question-heading"
          htmlFor="summary"
        >
          Summary
        </label>
        <p
          className="hint"
          id="summary-hint"
        >
          3-4 sentences that describe your business. This can be seen by all Digital Marketplace visitors without signing in.
        </p>
        <div>
          <textarea
            aria-describedby="summary-hint"
            className="under-50"
            defaultValue=""
            id="summary"
            name="summary"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
          />
          <span
            className="word-count-counter"
          >
            50 words remaining
          </span>
        </div>
      </div>
      <div
        className="field"
      >
        <label
          className="question-heading"
          htmlFor="website"
        >
          Website URL
        </label>
        <p
          className="hint"
          id="website-hint"
        >
          Provide a link to your website beginning with http
        </p>
        <input
          aria-describedby="website-hint"
          className=""
          disabled={false}
          id="website"
          maxLength={undefined}
          name="website"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          pattern={undefined}
          readOnly={undefined}
          type="text"
          value=""
        />
      </div>
      <div
        className="field"
      >
        <label
          className="question-heading"
          htmlFor="linkedin"
        >
          LinkedIn URL (optional)
        </label>
        <p
          className="hint"
          id="linkedin-hint"
        >
          Provide a LinkedIn URL beginning with http
        </p>
        <input
          aria-describedby="linkedin-hint"
          className=""
          disabled={false}
          id="linkedin"
          maxLength={undefined}
          name="linkedin"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          pattern={undefined}
          readOnly={undefined}
          type="text"
          value=""
        />
      </div>
      <div
        className="field"
      >
        <label
          className="question-heading"
          htmlFor="address_line"
        >
          Primary Address
        </label>
        <p
          className="hint"
          id="address_line-hint"
        >
          Principal place of business.
        </p>
        <input
          aria-describedby="address_line-hint"
          className=""
          disabled={false}
          id="address_line"
          maxLength={undefined}
          name="address.address_line"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          pattern={undefined}
          readOnly={undefined}
          type="text"
          value=""
        />
      </div>
      <div
        className="field"
      >
        <label
          className="question-heading"
          htmlFor="suburb"
        >
          Suburb
        </label>
        <input
          aria-describedby={undefined}
          className=""
          disabled={false}
          id="suburb"
          maxLength={undefined}
          name="address.suburb"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          pattern={undefined}
          readOnly={undefined}
          type="text"
          value=""
        />
      </div>
      <div
        className="field"
      >
        <label
          className="question-heading"
          htmlFor="state"
        >
          State
        </label>
        <input
          aria-describedby={undefined}
          className=""
          disabled={false}
          id="state"
          maxLength={undefined}
          name="address.state"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          pattern={undefined}
          readOnly={undefined}
          type="text"
          value=""
        />
      </div>
      <div
        className="field"
      >
        <label
          className="question-heading"
          htmlFor="postal_code"
        >
          Postcode
        </label>
        <input
          aria-describedby={undefined}
          className=""
          disabled={false}
          id="postal_code"
          maxLength="4"
          name="address.postal_code"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          pattern={undefined}
          readOnly={undefined}
          type="text"
          value=""
        />
      </div>
      <div>
        <p
          styleName="footer"
        >
          More offices?
        </p>
        <button
          className="button-secondary"
          onClick={[Function]}
          type="submit"
        >
          Add another address
        </button>
      </div>
      <div
        styleName="nav-row"
      >
        <button
          styleName="nav-button"
          type="submit"
        >
          Update profile
        </button>
        <button
          className="save-button"
          onClick={[Function]}
          styleName="nav-button"
        >
          Save and finish later
        </button>
        <button
          className="button-secondary"
          onClick={[Function]}
          styleName="nav-link skip-button"
        >
          Skip for now
        </button>
      </div>
    </form>
  </article>
</div>
`;
